src/settingsview.cpp

```diff
@@ -119,22 +113,27 @@ void SettingsView::browseMonitorDir()
     if (dir == monitorDir || dir.isEmpty()) {
         return;
     }
-    if (confirmDialogMonitorDir(dir, false)) {
+    if (confirmDialogMonitorDir(dir)) {
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
         ui->monitorClear->show();
-        KiwixApp::instance()->getLibrary()->loadMonitorDir(dir);
         KiwixApp::instance()->getWatcher()->addPath(dir);
+        auto previousDir = KiwixApp::instance()->getSettingsManager()->getMonitorDir();
```

Comment:
`previousDir` must be obtained *before* the call to `setMonitorDir()`

Reply:
Wow, that fixes something I was concerned about, thank you!